Client/Frontend/Login/LoginListViewController.swift

```diff
@@ -123,29 +144,29 @@ class LoginListViewController: LoginAuthViewController {
     }
     
     private func fetchLoginInfo(_ searchQuery: String? = nil) {
-        guard !isFetchingLoginEntries else {
-            return
-        }
-        
-        isFetchingLoginEntries = true
-        
-        if let query = searchQuery {
-            profile.logins.getLoginsForQuery(query) >>== { [weak self] results in
-                self?.reloadEntries(results: results)
-            }
-        } else {
-            profile.logins.getAllLogins() >>== { [weak self] results in
-                self?.reloadEntries(results: results)
+        if !isCredentialsRefreshing {
+            isCredentialsRefreshing = true
+                    
+            passwordAPI.getSavedLogins { credentials in
+                self.reloadEntries(with: searchQuery, passwordForms: credentials)
             }
         }
     }
     
-    private func reloadEntries(results: Cursor<Login>) {
-        loginEntries = results.asArray()
+    private func reloadEntries(with query: String? = nil, passwordForms: [PasswordForm]) {
+        if let query = query, !query.isEmpty {
+            credentialList = passwordForms.filter { form in
+                (form.url.origin?.lowercased() ?? "").contains(query) ||
+                (form.usernameValue?.lowercased() ?? "").contains(query)
```

Comment:
You seem to use many optional fallback in this pull request.
I would audit it to see in which places it's worth to keep it and in other places maybe convert to early returns,
reason I ask is that it may result in undefined behavior, empty cells etc.

Reply:
For this case it actually doesnt matter since we are using it while doing a search and adding an extra `query.isEmpty` satisfies the condition. However it makes more sense to implement this without a fallback t least for the sake of readability. 

Change at https://github.com/brave/brave-ios/pull/5013/commits/25f3d8d938c0ec02e77c7fa1a27e9ca9b275ce18